### PR TITLE
build: fix musl + add compiler.h

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by brauner** [comment]
jenkins: test this please

File: src/lxc/network.c [hunk 1]
```diff
@@ -2102,7 +2103,7 @@ static int lxc_create_network_unpriv_exec(const char *lxcpath, const char *lxcna
 	if (child == 0) {
 		int ret;
 		size_t retlen;
-		char pidstr[LXC_NUMSTRLEN64];
+		char pidstr[INTTYPE_TO_STRLEN(int)];
```
Comment (Blub) [inline src/lxc/network.c]: `int` -> `pid_t`

File: src/lxc/network.c [hunk 2]
```diff
@@ -2124,10 +2125,10 @@ static int lxc_create_network_unpriv_exec(const char *lxcpath, const char *lxcna
 			_exit(EXIT_FAILURE);
 		}
 
-		ret = snprintf(pidstr, LXC_NUMSTRLEN64, "%d", pid);
-		if (ret < 0 || ret >= LXC_NUMSTRLEN64)
+		ret = snprintf(pidstr, INTTYPE_TO_STRLEN(int), "%d", pid);
```
Comment (Blub) [inline src/lxc/network.c]: Maybe use `sizeof(pidstr)` instead?

File: src/lxc/network.c [hunk 3]
```diff
@@ -2124,10 +2125,10 @@ static int lxc_create_network_unpriv_exec(const char *lxcpath, const char *lxcna
...
+		if (ret < 0 || ret >= INTTYPE_TO_STRLEN(int))
```
Comment (Blub) [inline src/lxc/network.c]: `sizeof(pidstr)`

File: src/lxc/network.c [hunk 4]
```diff
@@ -2124,10 +2125,10 @@ static int lxc_create_network_unpriv_exec(const char *lxcpath, const char *lxcna
...
 			_exit(EXIT_FAILURE);
-		pidstr[LXC_NUMSTRLEN64 - 1] = '\0';
+		pidstr[INTTYPE_TO_STRLEN(int) - 1] = '\0';
```
Comment (Blub) [inline src/lxc/network.c]: `sizeof(pidstr)-1`

**Comment by brauner** [comment]
On Mon, Aug 27, 2018 at 01:41:17AM -0700, Wolfgang Bumiller wrote:
> Blub requested changes on this pull request.
> 
> I'm not a fan of `INTTYPE_TO_STRLEN()`, too easy to get the types wrong, especially in C where there's no compile-time connection between the arrays and the text. I'd rather just stick to some bigger number which is the same everywhere.

I like the idea of being as small as possible since we know that there
are a bunch of people and companies out there that run us on embedded
systems were size might really matter.
But you have to power to NAK this since you're maintainer now. :)

**Comment by Blub** [comment]
I guess it's fine. Just need to be careful. Often the code filling the buffer takes a size argument anyway.

**Comment by brauner** [comment]
Ok, I'm fixing this up now and then you can merge, @Blub.

File: src/lxc/network.c (hunk 1 above)
Comment (brauner) [reply src/lxc/network.c]: fixed

File: src/lxc/network.c (hunk 2 above)
Comment (brauner) [reply src/lxc/network.c]: fixed

**Pull request merged**